src/main/java/nl/tudelft/ewi/devhub/server/web/resources/ProjectAssignmentsResource.java

```diff
@@ -0,0 +1,312 @@
+        parameters.put("group", group);
+        parameters.put("course", group.getCourse());
+        parameters.put("repository", repository);
+        parameters.put("deliveries", deliveries);
+
+        List<Locale> locales = Collections.list(request.getLocales());
+        return display(templateEngine.process("courses/assignments/group-assignments.ftl", locales, parameters));
+    }
+
+    /**
+     * Get a specific assignment. Administrators and assistants see delivered assignments.
+     * Students see whether or not an assignment has been delivered, and their grades.
+     * @param request the current HttpServletRequest
+     * @param assignmentId assignmentId for the assignment
+     * @return rendered assignment overview
+     * @throws IOException if an I/O error occurs
+     */
+    @GET
+    @Transactional
+    @Path("{assignmentId : \\d+}")
+    public Response getAssignmentView(@Context HttpServletRequest request,
+                                      @PathParam("assignmentId") Long assignmentId)
+            throws IOException, ApiError, GitClientException {
+
+        Assignment assignment = assignments.find(group.getCourse(), assignmentId);
+        Repository repository = gitClient.repositories().retrieve(group.getRepositoryName());
+
+        Map<String, Object> parameters = Maps.newLinkedHashMap();
+        parameters.put("user", currentUser);
+        parameters.put("group", group);
+        parameters.put("course", group.getCourse());
+        parameters.put("repository", repository);
+        parameters.put("assignment", assignment);
+        parameters.put("myDeliveries", deliveries.getDeliveries(assignment, group));
+        parameters.put("states", new ProjectResource.CommitChecker(group, buildResults));
+        parameters.put("recentCommits", repository.retrieveBranch("master").retrieveCommits(0, 25).getCommits());
+
+        List<Locale> locales = Collections.list(request.getLocales());
+        return display(templateEngine.process("courses/assignments/group-assignment-view.ftl", locales, parameters));
+    }
+
+    /**
+     * Submit an assignment for a course
+     * @param request the current HttpServletRequest
+     * @param assignmentId assignmentId for the assignment
+     * @param formData submit data
+     * @return a redirect request to the assignment page
+     * @throws IOException if an I/O error occurs
+     * @throws ApiError if an ApiError occurs
+     */
+    @POST
+    @Path("{assignmentId : \\d+}")
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    public Response postAssignment(@Context HttpServletRequest request,
+                                   @PathParam("assignmentId") Long assignmentId,
+                                   MultipartFormDataInput formData) throws IOException, ApiError, GitClientException {
+
+        Map<String, List<InputPart>> formDataMap = formData.getFormDataMap();
+        String commitId = extractString(formDataMap, "commit-id");
```

Comment:
Deadlines should be enforced, not with exceptions. Deadlines shouldn't be too strict because they can be interpreted in multiple ways (eg. first deadline, second deadline, resubmission after a mistake, resubmission with penalty). But an assignment that was delivered too late, should be marked as such.

However, i'm still looking for a good way to fill the deadline fields, because `input=datetime-local` doesn't really have good browser support. Therefore we are not able to make these date comparisons yet.